remote-messaging/remote-messaging-api/src/main/java/com/duckduckgo/remote/messaging/api/RemoteMessage.kt

```diff
@@ -78,10 +90,18 @@ sealed class Content(val messageType: MessageType) {
     }
 }
 
-sealed class Action(val actionType: String, open val value: String) {
-    data class Url(override val value: String) : Action(URL.jsonValue, value)
-    data class PlayStore(override val value: String) : Action(PLAYSTORE.jsonValue, value)
-    object DefaultBrowser : Action(DEFAULT_BROWSER.jsonValue, "")
-    object Dismiss : Action(DISMISS.jsonValue, "")
-    object AppTpOnboarding : Action(APP_TP_ONBOARDING.jsonValue, "")
+sealed class Action(val actionType: String, open val value: String, open val additionalParameters: Map<String, String>?) {
+    data class Url(override val value: String) : Action(URL.jsonValue, value, null)
+    data class PlayStore(override val value: String) : Action(PLAYSTORE.jsonValue, value, null)
+    object DefaultBrowser : Action(DEFAULT_BROWSER.jsonValue, "", null)
+    object Dismiss : Action(DISMISS.jsonValue, "", null)
+    object AppTpOnboarding : Action(APP_TP_ONBOARDING.jsonValue, "", null)
+    data class Share(
+        override val value: String,
+        override val additionalParameters: Map<String, String>?,
+    ) : Action(SHARE.jsonValue, value, additionalParameters)
+
+    enum class AdditionalParameter(val key: String) {
```

Comment:
`additionalParameters` is a payload that can be anything for other actions. Knowing additionalParameter can contain a `title` is specific of the `Share` action, so it's an internal detail of that action.

You can move it inside: 
```
    data class Share(
        override val value: String,
        override val additionalParameters: Map<String, String>?,
    ) : Action(SHARE.jsonValue, value, additionalParameters) {

        val title: String
            get() = additionalParameters?.get(AdditionalParameter.TITLE.key).orEmpty()

        private enum class AdditionalParameter(val key: String) {
            TITLE("title"),
        }
    }
```

Reply:
I replaced it with your suggestion. However, if we ever have another action that requires the use of `TITLE` or any other additional parameter, I suppose we can have the enum class within the action to avoid duplications.